cram/cram_encode.c

```diff
@@ -87,6 +87,10 @@ cram_block *cram_encode_compression_header(cram_fd *fd, cram_container *c,
     cram_block *map = cram_new_block(COMPRESSION_HEADER, 0);
     int i, mc, r = 0;
 
+    pthread_mutex_lock(&fd->metrics_lock);
```

Comment:
Do you need to lock this (as you're accessing `c->no_ref` rather that `fd->no_ref`)?

Reply:
Haha indeed not!  I expect this was originally fd->no_ref and I blithly converted it to `c->` instead.

Reply:
You'll also need to check other places where it was changed, as there's at least one other case where you grab a lock before accessing `c->no_ref`.